flink-runtime/src/main/java/org/apache/flink/runtime/state/metrics/LatencyTrackingStateConfig.java

```diff
@@ -0,0 +1,117 @@
+public class LatencyTrackingStateConfig {
+
+    private final MetricGroup metricGroup;
+
+    private final boolean enabled;
+    private final int sampleInterval;
+    private final long slidingWindow;
+
+    LatencyTrackingStateConfig(
+            MetricGroup metricGroup, boolean enabled, int sampleInterval, long slidingWindow) {
+        this.metricGroup = metricGroup;
+        this.enabled = enabled;
+        this.sampleInterval = sampleInterval;
+        this.slidingWindow = slidingWindow;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public MetricGroup getMetricGroup() {
+        return metricGroup;
+    }
+
+    public long getSlidingWindow() {
+        return slidingWindow;
+    }
+
+    public int getSampleInterval() {
+        return sampleInterval;
+    }
+
+    public static LatencyTrackingStateConfig disabled() {
+        return newBuilder().setEnabled(false).build();
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private boolean enabled = StateBackendOptions.LATENCY_TRACK_ENABLED.defaultValue();
+        private int sampleInterval =
+                StateBackendOptions.LATENCY_TRACK_SAMPLE_INTERVAL.defaultValue();
+        private long slidingWindow =
+                StateBackendOptions.LATENCY_TRACK_SLIDING_WINDOW.defaultValue();
+        private MetricGroup metricGroup;
+
+        public Builder setEnabled(boolean enabled) {
+            this.enabled = enabled;
+            return this;
+        }
+
+        public Builder setSampleInterval(int sampleInterval) {
+            this.sampleInterval = sampleInterval;
+            return this;
+        }
+
+        public Builder setSlidingWindow(long slidingWindow) {
+            this.slidingWindow = slidingWindow;
+            return this;
+        }
+
+        public Builder setMetricGroup(MetricGroup metricGroup) {
+            this.metricGroup = metricGroup;
+            return this;
+        }
+
+        public Builder configure(ReadableConfig config) {
+            this.setEnabled(config.get(StateBackendOptions.LATENCY_TRACK_ENABLED))
+                    .setSampleInterval(
+                            config.get(StateBackendOptions.LATENCY_TRACK_SAMPLE_INTERVAL))
+                    .setSlidingWindow(config.get(StateBackendOptions.LATENCY_TRACK_SLIDING_WINDOW));
+            return this;
+        }
+
+        public LatencyTrackingStateConfig build() {
+            if (enabled) {
+                Preconditions.checkNotNull(
+                        metricGroup, "Metric group cannot be null if latency tracking is enabled.");
+                Preconditions.checkArgument(sampleInterval >= 1);
+            }
+            return new LatencyTrackingStateConfig(
+                    metricGroup, enabled, sampleInterval, slidingWindow);
```

Comment:
I'd move these check to constructor, as it potentially can be called from other places.